radish/documentors/__main__.py

```diff
@@ -0,0 +1,211 @@
+
+from radish.documentors.sphinx.multipage import MultiPageSphinxDocumentSet
+from radish.documentors.sphinx.singlepage import SinglePageSphinxDocument
+from radish.documentors.steps_writer import StepsDocumentWriter
+
+from radish.errororacle import error_oracle, enable_excepthook
+
+from radish.exceptions import RadishError
+
+from radish.loader import load_module_assets
+
+from radish.main import setup_config
+
+from radish.stepregistry import StepRegistry
+
+
+# ........................................................................... #
+@error_oracle
+def main():
+    usage = """
+Usage:
+    radish-docs sphinx <module-path>
+        [-o=<folder-path> | --output-folder=<folder-path>]
+        [--index-file-name=<file-path> ]
+        [-s=<file-path> | --single-file=<file-path>]
+        [-c | --print-to-console]
+        [--label-prefix=<label-prefix>]
+        [-r | --recreate]
+        [--no-ansi]
+    radish-docs (-h | --help)
+    radish-docs (-v | --version)
+
+Commands:
+    sphinx
+      create Sphinx documentation
+
+Arguments:
+    modules-path
+      path to python step modules to document
+
+Options:
+    -o=<folder-path>, --output-folder=<folder-path>
+      location of the output folder, by default `_build` in the current working
+      folder
+
+    --index-file-name=<file-name>
+      optional, name for the index file created by `--output-folder`, by
+      default "index.rst". This file is placed in the output folder.
+
+    -s=<file-path>, --single-file=<file-path>
+      write documentation to a single file
+
+    -c, --print-to-console
+      print `--single-file` documentaiton to console
+
+    --label-prefix=<label-prefix>
+      label prefix used for various section of the generated docs. Can be used
+      if you need a unique on in your documentation. By default set to
+      `radish.document_id.`
+
+    -r, --recreate
+      recreate output folder or file by removing and creating it again.
+
+    -h, --help
+      show this screen
+
+    -v, --version
+      show version
+
+(C) Copyright by Timo Furrer <tuxtimo@gmail.com>
+    """
+
+    # enable errororacles except hook system wide
+    enable_excepthook()
+
+    # parse sys.argv against our docopt argments
+    arguments = docopt("radish-docs {0}\n{1}".format(__VERSION__, usage),
+                       version=__VERSION__)
+
+    # create world wide configuration
+    setup_config(arguments)
+
+    # process arguments for "sphinx" command
+    if arguments['sphinx']:
+        # get the module path
+        module_path = arguments['<module-path>']
+
+        # get the output folder or if missing set the default for the
+        # output_folder_path
+        if arguments['--output-folder'] is not None:
+            output_folder_path = arguments['--output-folder']
+        else:
+            output_folder_path = "${PWD}/_build"
+
+        # get the index file name, otherwise default it to index.rst
+        if arguments['--index-file-name'] is not None:
```

Comment:
So the key of `--index-file-name` does exist in `arguments` and is set to None I think if the argument is not specified.
```
{'--help': False,
 '--index-file-name': None,
 '--label-prefix': None,
 '--no-ansi': False,
 '--output-folder': 'docs/radish/',
 '--print-to-console': True,
 '--recreate': False,
 '--single-file': None,
 '--version': False,
 '<module-path>': '.',
 'sphinx': True}
```
What kind of try/except are you envisioning if the value is None, because it would not be a KeyError.